src/conversions/chrono.rs

```diff
@@ -29,24 +29,38 @@ use crate::types::{
 use crate::{FromPyObject, IntoPy, PyAny, PyObject, PyResult, PyTryFrom, Python, ToPyObject};
 use chrono::offset::{FixedOffset, Utc};
 use chrono::{
-    div_mod_floor_64, DateTime, Datelike, Duration, NaiveDate, NaiveDateTime, NaiveTime, Offset,
-    TimeZone, Timelike, NANOS_PER_MICRO, SECS_PER_DAY,
+    DateTime, Datelike, Duration, NaiveDate, NaiveDateTime, NaiveTime, Offset, TimeZone, Timelike,
 };
 use std::convert::TryInto;
 
+const SECS_PER_DAY: i64 = 86_400;
+const MICROS_PER_SEC: i64 = 1_000_000;
+
 impl ToPyObject for Duration {
     fn to_object(&self, py: Python<'_>) -> PyObject {
-        let micros = self.nanos / NANOS_PER_MICRO;
-        let (days, secs) = div_mod_floor_64(self.secs, SECS_PER_DAY);
-        // Python will check overflow so even if we reduce the size
-        // it will still overflow.
-        let days = days.try_into().unwrap_or(i32::MAX);
-        let secs = secs.try_into().unwrap_or(i32::MAX);
+        // This is the total number of microseconds in the duration,
+        // unlike std::time::Duration we can't have only the fractional part,
+        // so we need to calculate it ourselves.
+        let total_micros = self
+            .num_microseconds()
+            // num_microseconds returns None on overflow
+            .unwrap_or(i64::MAX);
+        let total_secs = total_micros / MICROS_PER_SEC;
+        let micros = total_micros - total_secs * MICROS_PER_SEC;
+        let days = self.num_days();
+        let secs = total_secs - days * SECS_PER_DAY;
```

Comment:
I feel like the previous version is better, I don't really want to put in another layer, and instead prefer to just use the internal representation, but not sure what @djc or @davidhewitt thinks about this.